src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -433,16 +509,16 @@ internal static bool TryEncodeCodePoint(uint codePoint, Span<byte> utf8, int ind
         /// <returns>A count of bytes needed to store the UTF-8 representation of <paramref name="codePoint"/></returns>
         internal static int GetNumberOfEncodedBytes(uint codePoint)
         {
-            if (codePoint <= 0x7F)
+            if (codePoint <= Utf8OneByteLastCodePoint)
                 return 1;
 
-            if (codePoint <= 0x7FF)
+            if (codePoint <= Utf8TwoBytesLastCodePoint)
                 return 2;
 
-            if (codePoint <= 0xFFFF)
+            if (codePoint <= Utf8ThreeBytesLastCodePoint)
                 return 3;
 
-            if (codePoint <= 0x1FFFFF)
+            if (codePoint <= Utf8FourBytesLastCodePoint)
                 return 4;
```

Comment:
FYI: Utf8FourBytesLastCodePoint = 0x10FFFF, the previous GetNumberOfEncodedBytes() method was incorrect as it was using the constant value of 0x1FFFFF.